### PR TITLE
refactor(files): Use `@nextcloud/files` functions where possible and make files views cancelable

### LINKED ISSUE #45723: [Bug]: Numerical accounts show all files/folders in "Shared" status in the web UI
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

Use accounts with all numerical symbols (e.g. 11223344), after logging in, all files and folders will be displayed as shared, and the first symbol of the account will be in English letters, which is normal.
aaaaaaaa

### Steps to reproduce

1. Login to the backend
2. Add users
3. Set the user account to 112233
4. Other arbitrary
5. Login using 112233
6. View the file list, which shows that all files are shared

### Expected behavior

"Shared" should not appear

### Installation method

Community Manual installation with Archive

### Nextcloud Server version

29

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 8.3

### Web server

Apache (supported)

### Database engine version

MySQL

### Is this bug present after an update or on a fresh install?

Fresh Nextcloud Server install

### Are you using the Nextcloud Server Encryption module?

None

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

_No response_

### List of activated Apps

_No response_

### Nextcloud Signing status

_No response_

### Nextcloud Logs

_No response_

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by susnux**
* (by using nextcloud-files) Resolves: https://github.com/nextcloud/server/issues/45723

## Summary
1. Use the functions from `@nextcloud/files` where possible, replaces a lot of duplicated code in the files views.
2. Make the `getContents` functions cancelable to possibly save some network requests
3. Fix the favorites :star: view by adding a fake root like for recent etc, because you shall not be able to upload to the "favorites" root.

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by artonge — APPROVED** [review]

**Review by Pytal — APPROVED** [review]

File: apps/files/src/services/Files.ts [hunk 1]
```diff
@@ -3,68 +3,25 @@
...
-		mime: node.mime || 'application/octet-stream',
-		size: props?.size as number || 0,
-		permissions,
-		owner,
-		root: rootPath,
-		attributes: {
-			...node,
-			...props,
-			hasPreview: props?.['has-preview'],
-			failed: props?.fileid < 0,
```
Comment (skjnldsv) [inline apps/files/src/services/Files.ts:53]: Lost data :warning: 
https://github.com/nextcloud-libraries/nextcloud-files/blob/d170d8479d5434e727b14d91dbaba7dcfce5f06a/lib/dav/dav.ts#L179-L183

File: apps/files/src/services/Files.ts [hunk 2]
```diff
@@ -3,68 +3,25 @@
...
-	}
-
-	const props = node.props as ResponseProps
-	const permissions = davParsePermissions(props?.permissions)
-	const owner = (props['owner-id'] || userId).toString()
-
-	const source = generateRemoteUrl('dav' + rootPath + node.filename)
-	const id = props?.fileid < 0
-		? hashCode(source)
-		: props?.fileid as number || 0
```
Comment (skjnldsv) [inline apps/files/src/services/Files.ts:38]: Also lost in the refactor
Lost data :warning: 
https://github.com/nextcloud-libraries/nextcloud-files/blob/d170d8479d5434e727b14d91dbaba7dcfce5f06a/lib/dav/dav.ts#L179-L183

Comment (susnux) [reply apps/files/src/services/Files.ts:38]: #45860

File: apps/files/src/services/Files.ts (hunk 1 above)
Comment (susnux) [reply apps/files/src/services/Files.ts:53]: #45860

**Pull request merged**